### PR TITLE
expose the SchemaTypeAdapter since it is used in the Schema class.

### PR DESCRIPTION
**Pull request by albertshau**
Also adding methods to parse a json string into a Schema.

### CODE REVIEW

File: cdap-api/src/main/java/co/cask/cdap/api/data/schema/SchemaTypeAdapter.java [hunk 1]
```diff
@@ -14,9 +14,9 @@
  * the License.
  */
 
-package co.cask.cdap.internal.io;
+package co.cask.cdap.api.data.schema;
```
Comment (chtyim) [inline cdap-api/src/main/java/co/cask/cdap/api/data/schema/SchemaTypeAdapter.java]: Can you keep it in the internal package of the `cdap-api` module? It's not intended to be a public class.

File: cdap-api/src/main/java/co/cask/cdap/api/data/schema/SchemaTypeAdapter.java [hunk 2]
```diff
@@ -44,6 +44,7 @@
  *  </pre>
  * </p>
  */
+@Beta
```
Comment (chtyim) [inline cdap-api/src/main/java/co/cask/cdap/api/data/schema/SchemaTypeAdapter.java]: If you keep it in the internal package, no need to have `@Beta`.

File: cdap-api/src/main/java/co/cask/cdap/api/data/schema/SchemaTypeAdapter.java (hunk 1 above)
Comment (chtyim) [inline cdap-api/src/main/java/co/cask/cdap/api/data/schema/SchemaTypeAdapter.java]: Meaning just move the class from `cdap-common` to `cdap-api` but keeping the package the same 
https://github.com/caskdata/cdap/tree/develop/cdap-api/src/main/java/co/cask/cdap/internal/io

**Comment by albertshau** [comment]
https://builds.cask.co/browse/CDAP-DUT512

**Comment by chtyim** [comment]
LGTM

**Pull request merged**